e2e/utils/pages/homepage.page.ts

```diff
@@ -0,0 +1,64 @@
+import {
+	HERO_ANIMATION_CANVAS,
+	LOGIN_BUTTON,
+	TOKENS_SKELETONS_INITIALIZED
+} from '$lib/constants/test-ids.constant';
+import { type InternetIdentityPage } from '@dfinity/internet-identity-playwright';
+import { type Page } from '@playwright/test';
+import { HOMEPAGE_URL, LOCAL_REPLICA_URL } from '../constants/e2e.constants';
+
+export class Homepage {
+	protected readonly _page: Page;
+
+	constructor(page: Page) {
+		this._page = page;
+	}
+
+	protected async hideHeroAnimation(): Promise<void> {
+		await this._page
+			.getByTestId(HERO_ANIMATION_CANVAS)
+			.evaluate((element) => (element.style.display = 'none'));
+	}
+
+	get page(): Page {
+		return this._page;
+	}
+
+	async goto(): Promise<void> {
+		await this._page.goto(HOMEPAGE_URL);
+	}
+
+	async waitForLoggedOut(): Promise<void> {
+		await this.goto();
+		await this._page.getByTestId(LOGIN_BUTTON).waitFor();
+		await this.hideHeroAnimation();
+	}
+}
+
+export class HomepageLoggedIn extends Homepage {
+	private readonly _iiPage: InternetIdentityPage;
```

Comment:
I assumed you prefixed `_page` with `_` because it is `protected`, so no I'm confuse what `_` means?

If you want to distinguish protected and private you can also use `#` for the latest.

```
readonly #iiPage: InternetIdentityPage;
```

If it means "internal", given that we are using Typescript, I'm not sure it adds much value.

But again, maybe I am just not understanding the purpose of the prefix.

Reply:
That was a mixup from my side. Now, since we don't access neither `page` nor `iiPage` outside of the classes, I simplified it by just defining them as `readonly`.